xds/src/main/java/io/grpc/xds/LoadStatsManager.java

```diff
@@ -119,43 +119,41 @@ void removeLoadStats(String cluster, @Nullable String clusterService) {
     return res;
   }
 
+  // Introduced for testing.
   @VisibleForTesting
   interface LoadStatsStoreFactory {
     LoadStatsStore newLoadStatsStore(String cluster, String clusterService);
   }
 
   /**
-   * Interface for client side load stats store. An {@code LoadStatsStore} maintains load stats per
-   * cluster:cluster_service exposed by traffic director from a gRPC client's perspective,
-   * including dropped calls. Load stats for endpoints are aggregated in locality granularity
-   * while the numbers of dropped calls are aggregated in cluster:cluster_service granularity.
+   * Interface for client side load stats store. A {@link LoadStatsStore} instance holds the load
+   * stats for a cluster from an gRPC client's perspective by maintaining a set of locality
+   * counters for each locality it is tracking loads for.
    */
   interface LoadStatsStore {
 
     /**
      * Generates a report based on recorded load stats (including RPC counts, backend metrics and
      * dropped calls) for the interval since the previous call of this method.
      */
-    // TODO(chengyuanzhang): do not use proto type directly.
     ClusterStats generateLoadReport();
 
     /**
-     * Track load stats for endpoints in the provided locality. Only load stats for endpoints
-     * in tracked localities will be included in generated load reports.
+     * Adds tracking for load stats sent to the given {@code locality}. Returns the counter
+     * object responsible for tracking the client load stats to the given {@code locality}.
+     * Only load stats for tracked localities will be included in generated load reports.
      */
     ClientLoadCounter addLocality(Locality locality);
 
     /**
-     * Drop tracking load stats for endpoints in the provided locality. Load stats for endpoints
-     * in removed localities will no longer be included in future generated load reports after
+     * Drops tracking for load stats sent to the given {@code locality}. Load stats for removed
+     * localities will no longer be included in future generated load reports after
      * their currently recording stats have been fully reported.
      */
     void removeLocality(Locality locality);
 
     /**
      * Records a drop decision.
-     *
-     * <p>This method is thread-safe.
```

Comment:
Why remove this? Seems not correct. It needs be thread-safe.

Reply:
It is thread-safe. But just document it there.

Reply:
As usage of it relies on this contract, it should be documented on the interface.

Reply:
I think it's better to document on `LoadStatsStoreImpl`, the interface itself does not have any thread safety implications. Just like `Map` vs `ConcurrentMap`.

Reply:
It depends on how you use it. In a class, usage like `Map m = new ConcurrentMap()`, then you can use `m` and rely on the thread safety documented in  `ConcurrentMap`. But if in a class like `Map m = someAbstractOject.getAbstractMap()`, then you can not use `m` assuming it has thread-safety.

In `EdsLoadBalancer2`, `loadStatsStore` is declared as interface type. `loadStatsStore = xdsClient.addClientStats(cluster, edsServiceName)` is also giving an interface type.  But `loadStatsStore.recordDroppedRequest()` is called assuming thread-safety.

Reply:
Alright, put back some comment on the thread-safety guaranteed for implementations.

Anyway, the operation itself doesn't need to have thread-safety implications. Just like `Map` as an example, `put`/`get` operations does not have thread-safety implications, it is the API consumer's responsible to use it correctly.   The API consumer can use `put` or `get` without external synchronization in multi-thread context if it is given that the concrete implementation is actually a `ConcurrentMap`.